packages/tools/client-debug-view/src/RendererOptions.tsx

```diff
@@ -0,0 +1,165 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+import React from "react";
+
+import { SharedCounter } from "@fluidframework/counter";
+import { SharedMap } from "@fluidframework/map";
+import { SharedString } from "@fluidframework/sequence";
+import { SharedObjectCore } from "@fluidframework/shared-object-base";
+
+import {
+    AudienceMemberView,
+    AudienceMemberViewProps,
+    OpView,
+    OpViewProps,
+    SharedCounterView,
+    SharedMapView,
+    SharedStringView,
+} from "./components";
+
+// TODOs:
+// - Default rendering policies for SharedTree? PropertyDDS?
+//   - If so, we should consider using existing debug visualizations created for those DDS_s.
+
+/**
+ * Renders child data of some shared object.
+ *
+ * @remarks
+ *
+ * The child data can be a primitive or a
+ * {@link @fluidframework/core-interfaces#IFluidHandle | handle to another shared object}.
+ */
+export type RenderChild = (childObject: unknown) => React.ReactElement;
+
+/**
+ * Signature for data object view renderer callbacks.
+ */
+export type RenderSharedObject = (
+    sharedObject: SharedObjectCore, // TODO: is this the right type?
+    renderChild: RenderChild,
+) => React.ReactElement;
+
+/**
+ * The type of a shared object.
+ * Can be acquired via {@link @fluidframework/datastore-definitions#IChannelFactory.Type} field of
+ * your shared-object's factory class.
+ *
+ * TODO: can we do something better here?
```

Comment:
Looking for feedback / suggestions on this. Is there a better type we can match on here for differentiating DDS implementations?